drivers/dma/dma_mcux_lpc.c

```diff
@@ -188,10 +192,72 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 
 	LOG_DBG("channel is %d", p_handle->channel);
 
-	if (config->source_chaining_en && config->dest_chaining_en) {
-		LOG_DBG("link dma out 0 to channel %d", config->linked_channel);
-		/* Link DMA_OTRIG 0 to channel */
-		INPUTMUX_AttachSignal(INPUTMUX, 0, config->linked_channel);
+	/* Disconnecting the Otrig mux from the chaining channel */
+	if (!config->source_chaining_en && !config->dest_chaining_en) {
+		if (!k_sem_take(&is_otrig_being_configured, K_NO_WAIT)) {
+			for (int i = 0; i < ARRAY_SIZE(dma_data->linked_otrig_ch); ++i) {
+				if (dma_data->linked_otrig_ch[i] != -1 &&
+				   (config->linked_channel == dma_data->linked_otrig_ch[i] ||
+				    channel == dma_data->linked_otrig_ch[i])) {
+					dma_data->linked_otrig_ch[i] = -1;
+					break;
+				}
+			}
+			k_sem_give(&is_otrig_being_configured);
+		} else { /* k_sem_take(..); */
+			LOG_ERR("\n\rDMA attempted to configure an Otrig mux"
+				"\n\r\twhile the mux was in use.\n\r");
+			return -EACCES;
+		}
+	} else {
+		if (!k_sem_take(&is_otrig_being_configured, K_NO_WAIT)) {
+			LOG_DBG("link dma 0 channel %d with channel %d",
+				channel, config->linked_channel);
+			uint8_t is_otrig_available = 0;
+			uint8_t otrig_index = 0;
+
+			for (; otrig_index < ARRAY_SIZE(dma_data->linked_otrig_ch); ++otrig_index) {
+				if (dma_data->linked_otrig_ch[otrig_index] == -1) {
+					is_otrig_available = 1;
+					break;
+				}
+			}
+			if (!is_otrig_available) {
```

Comment:
this if statement and boolean is unnecessary, you can put this code under the if on line 220 and invert the condition, then assume the rest of this block is if the trig is available since you would be returned otherwise